src/Tickets/Commerce/Gateways/Stripe/Hooks.php

```diff
@@ -578,4 +579,88 @@ public function provide_defaults_for_hidden_fields( $value, $field_id, $field )
 	public function filter_admin_notices( $notices ) {
 		return $this->container->make( Gateway::class )->filter_admin_notices( $notices );
 	}
+
+	/**
+	 * Filter Stripe currency precision based on Stripe's specific requirements.
+	 *
+	 * @since TBD
+	 *
+	 * @param array  $currency_data The currency data from the map.
+	 * @param string $currency_code The currency code.
+	 * @param string $gateway       The gateway name.
+	 *
+	 * @return array The modified currency data.
+	 */
+	public function filter_stripe_currency_precision( $currency_data, $currency_code, $gateway ) {
+		// Only apply Stripe-specific logic for the Stripe gateway.
+		if ( 'stripe' !== $gateway ) {
+			return $currency_data;
+		}
+
+		// Apply Stripe's currency precision rules.
+		$stripe_precision = $this->get_stripe_precision( $currency_code, $currency_data['decimal_precision'] ?? 2 );
+
+		// Update the currency data with Stripe's precision.
+		$currency_data['decimal_precision'] = $stripe_precision;
+
+		return $currency_data;
+	}
+
+	/**
+	 * Get the appropriate precision for Stripe based on their currency requirements.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $currency_code The currency code.
+	 * @param int    $default_precision The default precision from currency data.
+	 *
+	 * @return int The precision to use for Stripe.
+	 */
+	private function get_stripe_precision( $currency_code, $default_precision ) {
```

Comment:
Then here, allow `null` passed and if it's null (or non-numeric. NOT 0!) use the property.